### ISSUE TITLE
Indesign and CorelDraw support

### ISSUE THREAD

**Issue by Alteregohr** [QL-Win/QuickLook#346]
This is a great utility. I have no problems, but a feature request. Is it possible to add support for Adobe Indesign .indd and CorelDraw .cdr files preview.

**Comment by xupefei** [comment]
I am afraid that support `.indd` files are impossible since there no 3rd-party library can open such a file.

For `.cdr`, I found https://stackoverflow.com/questions/4864765/convert-cdr-corel-draw-using-net which seems promising. If you have a programming experience you can try to implement it as a plugin.

**Comment by Alteregohr** [comment]
Thank you for replay. Unfortunately my programming skills are not that good. Keep up the good work!

**Comment by xupefei** [comment]
Let's see if there is any volunteer ...

**Comment by jeremyhart** [comment]
Here is a starting point, a exif CLI that can read Indesign .indd files: https://www.sno.phy.queensu.ca/~phil/exiftool/#supported

**Comment by mewkevin** [comment]
We want cdr preview support. Anyone who know programming?

**Comment by schroef** [comment]
> I am afraid that support `.indd` files are impossible since there no 3rd-party library can open such a file.
> 
> For `.cdr`, I found https://stackoverflow.com/questions/4864765/convert-cdr-corel-draw-using-net which seems promising. If you have a programming experience you can try to implement it as a plugin.

On osx I got an application which does allow me to see the previews added when saving InDesign files. By default osx can't preview these files, with this app you are able to see the amount of pages set in InDesign.

Would a link be useful.to you?

**Comment by andersonbruno02** [comment]
Indesign still have no suport?

**Comment by zhangkaihua88** [comment]
I created [QuickLook.Plugin.CorelDrawViewer-Thumbnail](https://github.com/zhangkaihua88/QuickLook.Plugin.CorelDrawViewer-Thumbnail), which preview cdr file base on thumbnail file in cdr file.

**Comment by schroef** [comment]
> Indesign still have no suport?

There is an app in the app store, it's i believe around $3. It works perfect but o ly on oaxt

**Comment by emako** [comment]
Any supporting about abobe files, just check https://github.com/QL-Win/QuickLook/issues/1516

**Comment by StackOverflow0x** [comment]
> Here is a starting point, a exif CLI that can read Indesign .indd files: https://www.sno.phy.queensu.ca/~phil/exiftool/#supported

Yes, exiftool is the way. I wrote a plugin for the Seer previewing app referencing this executable before, so hopefully something like this could work in QuickLook.
`"exec":"exiftool.exe \"*SEER_INPUT_PATH*\" -listItem 0 -pageimage -b -m -W \"*SEER_OUTPUT_PATH*.jpg\""`
So basically it just converts the first page to a JPEG and previews that.

**Comment by emako** [comment]
QuickLook.Plugin.ExifTool is currently under construction for the *.indd (InDesign) files.

**Comment by schroef** [comment]
@emako own that's nice!
Adobe states that it's already a native festure, but it simply doesnt work

**Comment by emako** [comment]
Adobe InDesign files (*.indd, *.indt) is supported by 3rd-party plugin.

https://github.com/QL-Win/QuickLook/wiki/Available-Plugins

https://github.com/emako/QuickLook.Plugin.ExifTool/releases

**Comment by StackOverflow0x** [comment]
> Adobe InDesign files (*.indd, *.indt) is supported by 3rd-party plugin.
> 
> https://github.com/QL-Win/QuickLook/wiki/Available-Plugins
> 
> https://github.com/emako/QuickLook.Plugin.ExifTool/releases

This seems to work for me! All good here.

**Comment by schroef** [comment]
It does work, but not for multi page. I still need to make some example files to help get this improved
